pkg/autoscaler/stats_reporter.go

```diff
@@ -27,63 +27,39 @@ import (
 	"go.opencensus.io/tag"
 )
 
-// Measurement represents the type of the autoscaler metric to be reported
-type Measurement int
-
-const (
-	// DesiredPodCountM is used for the pod count that autoscaler wants
-	DesiredPodCountM Measurement = iota
-	// RequestedPodCountM is used for the requested pod count from kubernetes
-	RequestedPodCountM
-	// ActualPodCountM is used for the actual number of pods we have
-	ActualPodCountM
-	// ObservedPodCountM is used for the observed number of pods we have
-	ObservedPodCountM
-	// StableRequestConcurrencyM is the average of requests count per observed pod in each stable window (default 60 seconds)
-	StableRequestConcurrencyM
-	// PanicRequestConcurrencyM is the average of requests count per observed pod in each panic window (default 6 seconds)
-	PanicRequestConcurrencyM
-	// TargetConcurrencyM is the desired number of concurrent requests for each pod
-	TargetConcurrencyM
-	// PanicM is used as a flag to indicate if autoscaler is in panic mode or not
-	PanicM
-)
-
 var (
-	measurements = []*stats.Float64Measure{
-		DesiredPodCountM: stats.Float64(
-			"desired_pods",
-			"Number of pods autoscaler wants to allocate",
-			stats.UnitNone),
-		RequestedPodCountM: stats.Float64(
-			"requested_pods",
-			"Number of pods autoscaler requested from Kubernetes",
-			stats.UnitNone),
-		ActualPodCountM: stats.Float64(
-			"actual_pods",
-			"Number of pods that are allocated currently",
-			stats.UnitNone),
-		ObservedPodCountM: stats.Float64(
-			"observed_pods",
-			"Number of pods that are observed currently",
-			stats.UnitNone),
-		StableRequestConcurrencyM: stats.Float64(
-			"stable_request_concurrency",
-			"Average of requests count per observed pod in each stable window (default 60 seconds)",
-			stats.UnitNone),
-		PanicRequestConcurrencyM: stats.Float64(
-			"panic_request_concurrency",
-			"Average of requests count per observed pod in each panic window (default 6 seconds)",
-			stats.UnitNone),
-		TargetConcurrencyM: stats.Float64(
-			"target_concurrency_per_pod",
-			"The desired number of concurrent requests for each pod",
-			stats.UnitNone),
-		PanicM: stats.Float64(
-			"panic_mode",
-			"1 if autoscaler is in panic mode, 0 otherwise",
-			stats.UnitNone),
-	}
+	desiredPodCountM = stats.Int64(
+		"desired_pods",
+		"Number of pods autoscaler wants to allocate",
+		stats.UnitNone)
+	requestedPodCountM = stats.Int64(
+		"requested_pods",
+		"Number of pods autoscaler requested from Kubernetes",
+		stats.UnitNone)
+	actualPodCountM = stats.Int64(
+		"actual_pods",
+		"Number of pods that are allocated currently",
+		stats.UnitNone)
+	observedPodCountM = stats.Float64(
```

Comment:
Why does this remain a Float64 while actual is an Int64?

Reply:
This is the average observed pod count during the scaling window. It could be `float64` value since it's average with `float64` weight. However I wonder whether we should cast it to int when reporting the metric since it is a count. WDYT?

Reply:
I am not super familiar with the metric so don't currently have a strong opinion on casting.

The description "Number of pods that are observed currently" is nearly identical to "Number of pods that are allocated currently". Is the actual pod count not averaged over a window?

Does it make sense for someone to be graphing actual versus observed? Will it be confusing if one of these is an int and one is a float?

Reply:
Created #2994. Let's discuss there so this PR won't be blocked. We can revision this specific metric later.